### PR TITLE
"Create your first snap" - corrections and improvements

### LINKED ISSUE #209: [Creating your first snap] “Getting started” language errors and non-sequiturs
[The “Getting started” step](https://tutorials.ubuntu.com/tutorial/create-your-first-snap#0) contains several language and logic problems, which I’ll report as a single issue to save time:

1. “ubuntu core” should be “Ubuntu Core” (twice).

1. There’s an “If you are running ubuntu core” section, but no “If you are _not_ running Ubuntu Core” section. Either there’s a section missing, or the heading needs rewording.

1. There’s no instructions on how to tell whether I _am_ running Ubuntu Core.

1. “Ubuntu Core is not a traditional Linux distribution: there is no "apt"…” suggests that most/all traditional Linux distributions have `apt`, which isn’t the case.

1. “there is no "apt", the system is read-only” is a run-on sentence. Probably the comma should be followed by “and”.

1. “develop and build snap in” is a number mismatch: it should be either “build a snap” or “build snaps”.

1. “How can you develop and build snap in such an environment then?” is a non-sequitur: there’s no hint of why that would be difficult.

1. “We have a special snap for this!” is fake excitement. Change the “!” to “.”

1. “We have a special snap for this! Enter the "classic" snap. This snap…” repeats “snap” distractingly often. Either delete the first sentence, or change the second sentence to something like “It’s called `classic`.”.

1. “We can there install” should be “There we can install”.

1. “thus … then” is redundant. Delete either the “thus” or the “then”.

1. People are unlikely to know what “exit the shell classically” means. Is there another way to exit? If not, maybe this should be something like “exit the shell as normal”.

1. “(we won't remind you about that in the following and others tutorial.” is a nesting error — the parenthesis should either be completely inside a single sentence, or contain one or more whole sentences.

1. “the following and others” is redundant. Probably this should be “this and others”.

1. “others tutorial” is a number mismatch. Probably this should be “other tutorials”.

1. “as first thing” is both incorrect (it’s the second step, not the first) and missing an article. Either “as the second step”, or more simply, just say “next”.

Finally, this is not an error, but the tutorial would be less condescending if the words “just” and “simply” were deleted throughout.

### PR DESCRIPTION
**Pull request by pmatulis**
resolves #209

### CODE REVIEW

File: tutorials/packaging/create-your-first-snap.md [hunk 1]
```diff
@@ -49,94 +47,105 @@ Survey
...
-### If you are running Ubuntu Core
+: Note: other GNU/Linux distributions are currently in the process of building packages for
+snapcraft.
 
-Ubuntu Core is not a traditional Linux distribution: there is no “apt”, the system is read-only. How can you develop and build snap in such an environment then? We have a special snap for this!
+### Installing the Classic snap
 
-Enter the “**classic**” snap. This snap will get us into a chroot, where we have a traditional Ubuntu distribution. We can there install more packages, change files projects and run snapcraft to build our snap. The **home** directory is shared between the container and ubuntu core. We can thus build a snap, then, outside of the classic snap, install and test it.
+You should be running Ubuntu Core for this. To develop and build a snap in such an environment we
+will use a special snap call the **classic** snap.
```
Comment (evilnick) [inline tutorials/packaging/create-your-first-snap.md]: called

File: tutorials/packaging/create-your-first-snap.md [hunk 2]
```diff
@@ -171,336 +192,416 @@ grade: devel
...
 
+To build **hello**, add the following to your `snapcraft.yaml` file:
 
-```bash
+```yaml
 parts:
   gnu-hello:
 	source: http://ftp.gnu.org/gnu/hello/hello-2.10.tar.gz
 	plugin: autotools
 ```
```
Comment (evilnick) [inline tutorials/packaging/create-your-first-snap.md]: the on screen version has the wrong indent (not valid yaml) probably replace these tabs(?)

Comment (pmatulis) [reply tutorials/packaging/create-your-first-snap.md]: Most of these YAML excerpts are like that. I don't know why this one stands out. Anyway, I went through them all (using 2-space indenting).

Also, only the initial PR gets a demo built.

File: tutorials/packaging/create-your-first-snap.md (hunk 1 above)
Comment (pmatulis) [reply tutorials/packaging/create-your-first-snap.md]: :white_check_mark:

**Comment by pmatulis** [comment]
Done.

**Comment by pmatulis** [comment]
I'm thinking that we should just make a mention of Ubuntu Core, instead of drilling into it (i.e. 'classic' snap, etc).

**Comment by jsseidel** [comment]
I like the idea about a mention of Ubuntu Core for the IoT folks who might read this instead of as the method for building. But, and I would need to play with it, it could be a stop-gap solution for building snaps in 18.04. I've made a little task for myself to try it.

**Comment by jsseidel** [comment]
Alas, this method didn't work for me in 18.04 (see below). I tried both edge and beta channels and the result was the same. If you want, I can track this down with the snap folks to see if the "classic" snap is deprecated or just has a bug.

```
$ sudo snap install classic --beta --devmode
classic (beta) 16.04 from 'canonical' installed
$ sudo classic
Creating classic environment
Parallel unsquashfs: Using 4 processors
11427 inodes (12760 blocks) to write

[=====================================================================================================================-] 12760/12760 100%

created 8870 files
created 1425 directories
created 2468 symlinks
created 79 devices
created 0 fifos
cp: cannot stat '/run/resolvconf/resolv.conf': No such file or directory
$
```

Then, subsequent attempts resulted in this:

```
$ sudo classic
sudo: unknown user: jsseidel
sudo: unable to initialize policy plugin
$
```

**Comment by pmatulis** [comment]
On Xenial I get:

```
[...]
Get:46 http://archive.ubuntu.com/ubuntu xenial-backports/universe Translation-en [3996 B]
Fetched 28.8 MB in 8s (3296 kB/s)
Reading package lists... Done
cp: cannot stat '/etc/writable/hostname': No such file or directory

$ sudo classic
sudo: unknown user: ubuntu
sudo: unable to initialize policy plugin
```

On Bionic I get the same as you.

I better remove the Ubuntu Core stuff completely.

**Pull request merged**